compiler/fir/cones/src/org/jetbrains/kotlin/fir/types/ConeTypes.kt

```diff
@@ -126,7 +126,7 @@ open class ConeFlexibleType(
         get() = lowerBound.nullability.takeIf { it == upperBound.nullability } ?: ConeNullability.UNKNOWN
 
     override val attributes: ConeAttributes
-        get() = lowerBound.attributes
+        get() = lowerBound.attributes.union(upperBound.attributes)
```

Comment:
It's designed that `FlexibleType.attributes == lowerBound.attributes`

Reply:
Thanks for the clarification. So, this attribute should be associated with lower bound. Reverted this part to the original expression.